### PR TITLE
[Network Drive]Add local extraction feature

### PR DESCRIPTION
**Pull request by akanshi-elastic**
#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

File: tests/sources/test_network_drive.py [hunk 1]
```diff
@@ -276,82 +275,68 @@ async def test_fetch_file_when_file_is_inaccessible(file_mock):
         file_mock.side_effect = SMBOSError(ntstatus=0xC0000043, filename="file1.txt")
 
         # Execute
-        response = source.fetch_file_content(path=path)
+        async for response in source.fetch_file_content(path=path):
+            # Assert
+            assert response is None
```
Comment (navarone-feekery) [inline tests/sources/test_network_drive.py:280]: I know it wasn't part of your PR specifically, but this test really should assert more than `response is None`. Can you add a check to make sure the expected logger is output?

**Review by navarone-feekery — COMMENTED** [review]
Looks good! I've left one request and one question before approving.

File: tests/sources/test_network_drive.py [hunk 2]
```diff
@@ -276,82 +275,68 @@ async def test_fetch_file_when_file_is_inaccessible(file_mock):
...
-            "Mock....", "utf-8"
-        )
-
         mock_response = {
             "id": "1",
             "_timestamp": "2022-04-21T12:12:30",
             "title": "file1.txt",
             "path": "\\1.2.3.4/Users/folder1/file1.txt",
-            "size": "50",
+            "size": 50,
```
[the remark this answers is not in the document]
Comment (akanshi-elastic) [reply tests/sources/test_network_drive.py:302]: Yes, response was integer only, might be it was mistake earlier.

**Review by navarone-feekery — APPROVED** [review]
LGTM! 🚀

**Pull request merged**